src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointClusterer.java

```diff
@@ -1,61 +1,70 @@
 package org.broadinstitute.hellbender.tools.spark.sv;
 
+import org.apache.spark.api.java.function.FlatMapFunction;
+import org.apache.spark.api.java.function.Function;
+
 import java.util.*;
-import java.util.function.Function;
 
 /**
  * A class that acts as a filter for breakpoint evidence.
  * It passes only that evidence that is part of a putative cluster.
  */
-public final class BreakpointClusterer implements Function<BreakpointEvidence, Iterator<BreakpointEvidence>> {
+public final class BreakpointClusterer implements Iterator<BreakpointEvidence> {
     private final int minEvidenceCount;
     private final int staleEventDistance;
-    private final SortedMap<BreakpointEvidence, Boolean> locMap = new TreeMap<>();
-    private final List<Map.Entry<BreakpointEvidence, Boolean>> reportableEntries;
-    private static final Iterator<BreakpointEvidence> noEvidence = Collections.emptyIterator();
-    private int currentContig = -1;
+    private final SVIntervalTree<List<BreakpointEvidence>> evidenceTree;
+    private Iterator<SVIntervalTree.Entry<List<BreakpointEvidence>>> treeItr;
+    private Iterator<BreakpointEvidence> listItr;
 
     public BreakpointClusterer( final int minEvidenceCount, final int staleEventDistance ) {
         this.minEvidenceCount = minEvidenceCount;
         this.staleEventDistance = staleEventDistance;
-        this.reportableEntries = new ArrayList<>(2 * minEvidenceCount);
+        this.evidenceTree = new SVIntervalTree<>();
     }
 
-    @Override
-    public Iterator<BreakpointEvidence> apply( final BreakpointEvidence evidence ) {
-        if ( evidence.getContigIndex() != currentContig ) {
-            currentContig = evidence.getContigIndex();
-            locMap.clear();
+    public Iterator<BreakpointEvidence> apply( final Iterator<BreakpointEvidence> evidenceItr ) {
```

Comment:
Good point.  I've renamed 'apply' to 'buildTree', made it void and private, and called it from the constructor.